pkg/cmd/grafana-cli/commands/datamigrations/encrypt_datasource_passwords_test.go

```diff
@@ -20,19 +20,30 @@ func TestPasswordMigrationCommand(t *testing.T) {
 	datasources := []*models.DataSource{
 		{Type: "influxdb", Name: "influxdb", Password: "foobar"},
 		{Type: "graphite", Name: "graphite", BasicAuthPassword: "foobar"},
-		{Type: "prometheus", Name: "prometheus", SecureJsonData: securejsondata.GetEncryptedJsonData(map[string]string{})},
+		{Type: "prometheus", Name: "prometheus"},
+		{Type: "elasticsearch", Name: "elasticsearch", Password: "pwd"},
 	}
 
 	// set required default values
 	for _, ds := range datasources {
 		ds.Created = time.Now()
 		ds.Updated = time.Now()
-		ds.SecureJsonData = securejsondata.GetEncryptedJsonData(map[string]string{})
+		if ds.Name == "elasticsearch" {
+			ds.SecureJsonData = securejsondata.GetEncryptedJsonData(map[string]string{
+				"key": "value",
+			})
+		} else {
+			ds.SecureJsonData = securejsondata.GetEncryptedJsonData(map[string]string{})
+		}
 	}
 
 	_, err := session.Insert(&datasources)
 	assert.Nil(t, err)
 
+	// force secure_json_data to be null to verify that migration can handle that
+	_, err = session.Exec("update data_source set secure_json_data = null where name = 'influxdb'")
```

Comment:
Could this be moved into the insert part and just not set the `ds.SecureJsonData` for influx (adding `if` next to `if ds.Name == "elasticsearch" {`) or is there some caveat why is this done in extra sql?

Reply:
I tried to use the insert, but couldn't get it to insert null for secure_json_data, it was always set to empty string. Don't understand how certain Grafana instances could have null in the first place for secure_json_data though. Could also be related to historic xorm upgrades maybe handling 
null values differently?

Going to try and use the api and create a datasource and see if I can force a null that way.

Reply:
Right with xorm you have to set something like `session.MustCols("secure_json_data")` to mark that if it is zero value in the insert it will be null. I assume something like this could have changed between some versions.

Reply:
Looked into using `session.MustCols("secure_json_data")`and it seemed to work at first, but behaved very weird. 

In the current state of this pr when updating a datasource secure_json_data with an explicit null value and inspecting the mysql database directly after the update have been written the secure_json_data column has a value of `NULL`. This is an actual NULL value in the database.

Using `session.MustCols("secure_json_data")` and inspecting the mysql database directly after the inserts I can see that secure_json_data column has a value of `null`. This is not an actual NULL value in the database, instead it's a string with the value `null`.

Because of this me and @aocenas decided that we'll merge these changes for the 6.3-beta1 and let users report if there should be any problem or some scenario left to handle.